### ISSUE TITLE
ability to make video title as filename instead of a constant "video title"

### ISSUE THREAD

**Issue by ahmedtds** [porjo/youtubeuploader#82]
Hi

is it possible to set video title as filename instead of a constant "video title" title?

without using jsons etc.. with only command line.

if not can you add such an option please?

**Comment by Jaom** [comment]
https://github.com/porjo/youtubeuploader
Usage specifies that there is an optional `-title` parameter which you can use.

**Comment by ahmedtds** [comment]
> 
> 
> https://github.com/porjo/youtubeuploader
> Usage specifies that there is an optional `-title` parameter which you can use.

Yes but can that be automated instead of typing it manually? Maybe default behaviour being like this would be more sensible?

**Comment by Jaom** [comment]
Calling a program (youtubeuploader) with a parameter of your choice is trivial to automate in every environment. 
I'm also very confused about your idea of the proper default behavior. Currently `youtubeuploader` checks whether there is `-title "Super Title For The Film"` if not then it looks up `.json` file (if specified) and if there is no title provided by the user via one of those two ways it uses some default value (`Video title`).

`youtubeuploader` is not an interactive program - you call it with a set of arguments and it does its job. The only interactive part is, I believe, the headless authorization fired the first time.

**Comment by ahmedtds** [comment]
Yes I hear you, I'm just saying since setting filename property is bugged (https://github.com/porjo/youtubeuploader/issues/73) maybe it may be a good idea to have the default value of title as filename provided instead of a constant like "video title"

**Comment by jorisrobijn** [comment]
#73 is about the filename in youtube. Not about the title in youtube. But the filename property in youtube is normally not used anywhere afaik. 
Do you mean that the title could by default be the specified file name if there is no title is on the command line AND none is in the json file? For automation this is a non-issue, you could add the same thing as filename and title. For manual upload it is indeed an inconvenience that you have to specify the file name twice.

**Comment by ahmedtds** [comment]
> Do you mean that the title could by default be the specified file name if there is no title is on the command line AND none is in the json file?
Yes, exactly that's what I mean, if none specified make it the filename instead of a constant "video title", that would be better imo.

> #73 is about the filename in youtube. Not about the title in youtube.
Yes, I know. I said that because think about it, you have several videos uploaded all with the same title "video title", you want to look that filename field in youtube to remember a detail about it, but it says "unknown" because it's bugged, and title is "video title". You say "damn I wish I had set video title something else like filename or something".

**Comment by SlavkoPavlovic** [comment]
Not sure why this doesn't get added, this is possible to do just by adding fours lines of code: 
```
var file = filepath.Base(*filename)
 var extension = filepath.Ext(file)
 file = file[0:len(file)-len(extension)]
 *title = file
```
I added this

**Comment by SlavkoPavlovic** [comment]
Not sure why this doesn't get added by default, this is possible to do by adding an if statement, four lines of code and editing one line:
```
if *title == "" {
	var file = filepath.Base(*filename)
	var extension = filepath.Ext(file)
	file = file[0:len(file)-len(extension)]
	*title = file
        }
```
From:
          `title = flag.String("title", "", "video title")`
To: 
          `title = flag.String("title", "video title", "video title")`

**Comment by porjo** [comment]
Fixed in #84

**Comment by ahmedtds** [comment]
Thank you for implementing this <3
